pkg/contractreader/extended.go

```diff
@@ -89,6 +124,26 @@ func (e *extendedContractReader) getOneBinding(contractName string) (ExtendedBou
 	}
 }
 
+func (e *extendedContractReader) QueryKey(
+	ctx context.Context,
+	contract types.BoundContract,
+	filter query.KeyFilter,
+	limitAndSort query.LimitAndSort,
+	sequenceDataType any,
+) ([]types.Sequence, error) {
+	if e.hasFinalityViolation() {
+		return nil, ErrFinalityViolated
```

Comment:
I am wondering why we want to revert finality violations, is it a product team requirement? 

Also how this is going to be used upstream?

Reply:
The plugins would fail to produce observations for any chain with a finality violation because the reader starts returning errors. My assumption is that many of these would bubble up through the plugin and generate logs. It might be worth adding some specific checks to the should transmit function as well.